tensordict/utils.py

```diff
@@ -1721,3 +1721,12 @@ def _legacy_lazy(func):
         )
     func.LEGACY = True
     return func
+
+
+# Process initializer for map
+def _proc_init(base_seed, queue):
+    worker_id = queue.get()
+    seed = base_seed + worker_id
+    torch.manual_seed(seed)
+    np_seed = _generate_state(base_seed, worker_id)
+    np.random.seed(np_seed)
```

Comment:
*If* `worker_init_fn` is exposed, then this seeding should happen *before* `worker_init_fn` gets called, so that users can override the seeds that are being used. (See https://github.com/pytorch/pytorch/blob/8ec59d3553db4e3dae991db9aa3251558265dfd3/torch/utils/data/_utils/worker.py#L250C35-L250C35)

`worker_init_fn` just takes `worker_id` as input, but users can call [`get_worker_info`](https://pytorch.org/docs/stable/data.html#torch.utils.data.get_worker_info) to retrieve more stuff like the seed they were seeded with (not sure how that would translate for TensorDict though).